### PR TITLE
Run CI on re-rendering PRs

### PR DESCRIPTION
**Pull request by jakirkham**
While it was initially a good idea to disable CI on re-rendering PRs as so many were being sent out (as so much was out-of-date), it is beginning not to seem like the case any more. Feedstocks often have to pin to particular versions of `conda-build` or have other custom workarounds to avoid new bugs that affect the ability to build the desired package. As a result, it is easy to miss why this was done in a PR or whether it is still needed without having CI. Also, it often happens that changes in the underlying stack affect such simple PRs. Enabling CI gives us a heads-up so we can react accordingly. This PR removes the skip CI comment in the message so that we can see if a re-rendering PR is acceptable and correct things accordingly.

cc @pelson @ocefpaf

### CODE REVIEW

File: scripts/regenerate_feedstock.py [hunk 1]
```diff
@@ -163,7 +163,7 @@ def create_update_pr(clone, remote_head, fork_remote, upstream_remote):
         clone.git.add('-A')
         from git import Actor
         author = Actor("conda-forge-admin", "pelson.pub+conda-forge@gmail.com")
-        commit = clone.index.commit("MNT: Updated the feedstock for conda-smithy version {}.\n\n[ci skip]".format(conda_smithy.__version__),
+        commit = clone.index.commit("MNT: Updated the feedstock for conda-smithy version {}.\n\n".format(conda_smithy.__version__),
```
Comment (pelson) [inline scripts/regenerate_feedstock.py]: The newlines would be fine to remove too.

Comment (jakirkham) [inline scripts/regenerate_feedstock.py]: Right, I guess they don't serve any purpose now.

Comment (jakirkham) [inline scripts/regenerate_feedstock.py]: Done.

**Comment by pelson** [comment]
+0 from me - even if the CI passes, there is no guarantee that it will pass on merge if things like new versions of conda, conda-build and soon conda-forge-build-setup are made available. By the same token, just because a feedstock previously built fine, there is no guarantee that it continues to build fine as the stack moves on. We will suffer from that more when `conda-forge-build-setup` is used - we will be able to make significant changes to that package (e.g. conda-build versions) without needing the feedstocks to be updated at all, and so will be even more susceptible to kinks in conda-build/conda.

In short - we would still be in a position of having some feedstocks that end up not building.

**Comment by jakirkham** [comment]
The thing is these PRs normally get people's attention who are otherwise busy maintaining code elsewhere. If we give them a false positive and merge, they may go back to what they are doing and completely miss the failure. However, if they get a failure, they won't merge the PR and will ping us. Even if they were not adding custom lines to the CI, knowing a problem is emerging like what happened with the `vs_runtime` package is helpful. Having more chances to catch these issues is a good thing. In the end, I think this is going to help us become aware of and track down problems faster.

**Comment by pelson** [comment]
You do make a convincing argument. In order to avoid x2 CI resources, how about we skip the CI on merge?

**Comment by jakirkham** [comment]
That seems reasonable. How should we go about implementing that?

**Comment by pelson** [comment]
> That seems reasonable. How should we go about implementing that?

Can we get that behaviour automatically by setting the title to include ``[no ci]`` (as in, does the merge commit message come from the PR title?)

**Comment by jakirkham** [comment]
Are you meaning `[skip ci]`? I wasn't sure what the answer was to that until a couple of days ago. Now it seems like the answer is yes after looking at GitHub's merge refs. Will update.

**Comment by jakirkham** [comment]
Updated. Please let me know if you need anything else.

**Comment by jakirkham** [comment]
Thoughts on this, @pelson? It matches your suggestion and does seem to be a viable solution.

**Comment by jakirkham** [comment]
Another good reason for doing this is that the time when the CI runs occur shifts from the time when people merge them to the time when the maintenance job generates. As the maintenance job tends to do this at low activity hours, it is less likely to conflict with users. However, the same cannot be said of running on merge.

**Comment by pelson** [comment]
This is easy to back out if we find we are flooding the CI services (GDAL, scipy, et. al - I'm looking at you 😄 ).

**Comment by jakirkham** [comment]
Thanks @pelson. Was the bot rebuilt with this?

**Comment by jakirkham** [comment]
I manually re-rendered some of the painful feedstocks tonight just to get them out of the way.

**Comment by pelson** [comment]
Yep, rebuilt - v1.3.1 is out of the door. Thanks for kicking off the painful builds. 👍

**Comment by jakirkham** [comment]
Of course.

Also as FYI this seems to do find on CircleCI and AppVeyor. However, Travis CI does an odd thing and tries to rebuild the last commit in the PR. I have raised this issue ( https://github.com/travis-ci/travis-ci/issues/6669 ) with them explaining this in more detail and providing an example.

**Comment by jakirkham** [comment]
Of course.

Also as FYI this seems to do fine on CircleCI and AppVeyor. However, Travis CI does an odd thing and tries to rebuild the last commit in the PR that proceeds the merge. I have raised this issue ( https://github.com/travis-ci/travis-ci/issues/6669 ) with Travis CI explaining this issue in more detail and providing an example.

**Comment by jakirkham** [comment]
Will need to keep an eye on PR ( https://github.com/travis-ci/travis-ci/issues/5374 ), which will break this workflow. In that issue, the OP has asked that CI be skipped for all commits in the PR if `[ci skip]` is in the title. This would block us from seeing if a re-rendering works on the PR and not running it on merge.
